src/tilearea.cpp

```diff
@@ -117,6 +117,25 @@ bool OrthogonalTileArea::Contains(TileIndex tile) const
 	return IsInsideBS(tile_x, left, this->w) && IsInsideBS(tile_y, top, this->h);
 }
 
+/**
+ * Expand a tile area by rad tiles in each direction, keeping within map bounds.
+ * @param rad Number of tiles to expand
+ */
+void OrthogonalTileArea::Expand(int rad)
```

Comment:
might simplify some code if this returns `OrthogonalTileArea &`/`this` ?

Reply:
Yea. I'll do that.

Reply:
not really? enables method chaining, is all

Reply:
Hmm, this doesn't seem to help any of the changes in this PR.

Reply:
could use `auto ta = ...` to remove the duplicated type. Up to you, but I think if you don't use method chaining, there's no point giving Expand a return type

Reply:
I kinda dislike using auto, except for complex template-iterator types. So...